packages/xstate-store/src/store.ts

```diff
@@ -253,85 +285,40 @@ export type TransitionsFromEventPayloadMap<
  * // Logs { context: { count: 5 }, status: 'active', ... }
  * ```
  */
-export function createStore<
+function _createStore<
   TContext extends StoreContext,
   TEventPayloadMap extends EventPayloadMap,
-  TTypes extends { emitted?: EventObject }
->({
-  context,
-  on,
-  types
-}: {
-  context: TContext;
-  on: {
-    [K in keyof TEventPayloadMap & string]:
-      | StoreAssigner<
-          NoInfer<TContext>,
-          { type: K } & TEventPayloadMap[K],
-          Cast<TTypes['emitted'], EventObject>
-        >
-      | StorePropertyAssigner<
-          NoInfer<TContext>,
-          { type: K } & TEventPayloadMap[K],
-          Cast<TTypes['emitted'], EventObject>
-        >;
-  };
-} & { types?: TTypes }): Store<
-  TContext,
-  ExtractEventsFromPayloadMap<TEventPayloadMap>,
-  Cast<TTypes['emitted'], EventObject>
->;
-
-/**
- * Creates a **store** that has its own internal state and can be sent events
- * that update its internal state based on transitions.
- *
- * @example
- *
- * ```ts
- * const store = createStore(
- *   // Initial context
- *   { count: 0 },
- *   // Transitions
- *   {
- *     inc: (context, event: { by: number }) => {
- *       return {
- *         count: context.count + event.by
- *       };
- *     }
- *   }
- * );
- *
- * store.subscribe((snapshot) => {
- *   console.log(snapshot);
- * });
- *
- * store.send({ type: 'inc', by: 5 });
- * // Logs { context: { count: 5 }, status: 'active', ... }
- * ```
- */
-export function createStore<
-  TContext extends StoreContext,
-  TEventPayloadMap extends EventPayloadMap
+  TEmitted extends EventPayloadMap
 >(
-  initialContext: TContext,
-  transitions: TransitionsFromEventPayloadMap<
-    TEventPayloadMap,
+  ...[{ context, on }]: CreateStoreParameterTypes<
     TContext,
-    EventObject
+    TEventPayloadMap,
+    TEmitted
   >
-): Store<TContext, ExtractEventsFromPayloadMap<TEventPayloadMap>, EventObject>;
-
-export function createStore(initialContextOrObject: any, transitions?: any) {
-  if (transitions === undefined) {
-    return createStoreCore(
-      initialContextOrObject.context,
-      initialContextOrObject.on
-    );
-  }
-  return createStoreCore(initialContextOrObject, transitions);
+): CreateStoreReturnType<TContext, TEventPayloadMap, TEmitted> {
+  return createStoreCore(context, on);
 }
 
+export const createStore: {
```

Comment:
If you are going with "Type parameters should now be explicitly provided to `createStore` and `createStoreWithProducer`" then those overloads are not needed at all here and this could be just simplified/inlined